component/loki/source/file/file.go

```diff
@@ -36,9 +37,32 @@ const (
 // Arguments holds values which are used to configure the loki.source.file
 // component.
 type Arguments struct {
-	Targets   []discovery.Target  `river:"targets,attr"`
-	ForwardTo []loki.LogsReceiver `river:"forward_to,attr"`
-	Encoding  string              `river:"encoding,attr,optional"`
+	Targets             []discovery.Target  `river:"targets,attr"`
+	ForwardTo           []loki.LogsReceiver `river:"forward_to,attr"`
+	Encoding            string              `river:"encoding,attr,optional"`
+	DecompressionConfig DecompressionConfig `river:"decompression,block,optional"`
+}
+
+type DecompressionConfig struct {
+	Enabled      bool          `river:"enabled,attr"`
+	InitialDelay time.Duration `river:"initial_delay,attr,optional"`
+	Format       string        `river:"format,attr"`
```

Comment:
One alternative way you could do this is by introducing a new type here for the format which implements encoding.TextMarshaler/encoding.TextUnmarshaler for validation. 

This would mean that if someone provides an invalid format, just the `format` attribute is highlighted as being a problem instead of the surrounding block. 

This is used in local.file for the `detector` attribute: https://github.com/grafana/agent/blob/9756a4ccc08fa33ab7006fed9fe5dfb44c0dc8ca/component/local/file/detector.go#L15-L65

I don't think you _have_ to do this, and it tends to be a bit more code, but just wanted to point it out as an option.

Reply:
> So maybe we should consider allowing the Validate to provide a name of attribute or block that is a problem?

I'd definitely be interested in something like this.